### ISSUE TITLE
loginLog.json format incorrect / standard fail2ban regex

### ISSUE THREAD

**Issue by ndom91** [causefx/Organizr#797]
###### Operating System: Ubuntu
<hr>

##### Problem Description:
organizr logs failed and successful logins incorrectly

<hr>

##### Reproduction Steps:
Manually clone organizr to webroot
Thats it..

<hr>

#### Errors on screen?  If so paste here:
So my loginLog.json output looks like so:
```
date":"2018-03-09 10:28:57","username":"[username]","ip":"1.2.3.4","auth_type":"bad_auth"},{"
```
It looks like it is not creating a new line at the end at the correct spot, instead it appends the ' {" ' which should be beginning a new line, onto the end of the last line.

This can be corrected for in fail2ban if you just add "..." at the end of the regex. (... being match and 3 characters in regex).

But still, this should be corrected. Where exactly does organizr write logs? I can look into it, I just cant seem to find it myself.

Thanks!

**Comment by causefx** [comment]
we had to do it for fail2ban for some reason it wouldnt read the json unless it was spilt line by line

**Comment by ndom91** [comment]
@causefx yeah thats all fine and good. However, it is not splitting the lines at the correct place. Therefore I had to adapt the fail2ban regex to be able to even read this output. 

You can see in my log output above at the end there is an extra ```{"``` as if it were starting a new line, but its still on the previous line. 

In short, it seems to be starting the new line two characters too late.

**Comment by ndom91** [comment]
so I've looked at that PR and the hack that made the made it print a new line, and i think I've found the issue. I just need to come up with a fix for it. So write write the new log entry on a new line, he simply str_replaced the variable `date` with `\r\ndate` - so he is dropping down to a new line directly before the date variable name, not including any of the formating characters before that. 

Which makes perfect sense with what we see:

`
date":"2018-03-09 10:28:57","username":"[username]","ip":"1.2.3.4","auth_type":"bad_auth"},{"
date":"2018-03-09 10:28:57","username":"[username]","ip":"1.2.3.4","auth_type":"bad_auth"},{"
date":"2018-03-09 10:28:57","username":"[username]","ip":"1.2.3.4","auth_type":"bad_auth"},{"`

should be 

`
{"date":"2018-03-09 10:28:57","username":"[username]","ip":"1.2.3.4","auth_type":"bad_auth"},
{"date":"2018-03-09 10:28:57","username":"[username]","ip":"1.2.3.4","auth_type":"bad_auth"},
{"date":"2018-03-09 10:28:57","username":"[username]","ip":"1.2.3.4","auth_type":"bad_auth"},`

TL;DR - Don't undo the PR #41 it just needs to be adjusted a bit. I'll try to come up with a solution and post back here as soon as possible.

**Comment by ndom91** [comment]
I'm not that familiar with fail2ban. I can look into it.

Currently, as I mentioned in my original post, you can simply add three dots `...` to the end of the fail2ban regex for it to match on any characters at the end of the line and this adaptation totally works with the current way it prints failed login lines. 

I can look into fail2ban json support though. 

Here's some basic fail2ban filter info: https://www.fail2ban.org/wiki/index.php/MANUAL_0_8#Usage

From that it soundsss like the regex filter mechanism detects/defines new events simply by new lines in the log. I'm sure others have tried to set a fail2ban filter on a json file before though. I'll do some more googling.

**Comment by causefx** [comment]
For now, if you like you can edit the Wiki to reflect the change:

https://github.com/causefx/Organizr/wiki/Fail2Ban-Integration

Closing this for now.
